collector/agent_poller.go

```diff
@@ -69,80 +61,120 @@ type AgentState struct {
 	executorAppNames map[string]string
 }
 
+type Agent struct {
+	AgentIP        string
+	IPCommand      string
+	Port           int
+	PollPeriod     int
+	Topic          string
+	AgentStateChan chan<- *AgentState
+}
+
+func NewAgent(
+	ipCommand string,
+	port int,
+	pollPeriod int,
+	topic string) (Agent, error) {
+	a := Agent{}
+	if len(ipCommand) < 0 {
+		return a, errors.New("Must pass ipAddress to NewAgent()")
+	}
+	if port < 1024 {
+		return a, errors.New("Must pass port to NewAgent()")
+	}
+	if pollPeriod == 0 {
+		return a, errors.New("Must pass pollPeriod to NewAgent()")
+	}
+	if len(topic) < 1 {
+		return a, errors.New("Must pass topic to NewAgent()")
+	}
+
+	a.IPCommand = ipCommand
+	a.Port = port
+	a.PollPeriod = pollPeriod
+	a.Topic = topic
+	a.AgentStateChan = make(chan *AgentState)
+
+	return a, nil
+}
+
 // Runs an Agent Poller which periodically produces data retrieved from a local Mesos Agent.
 // This function should be run as a gofunc.
-func RunAgentPoller(recordsChan chan<- *AvroDatum, agentStateChan chan<- *AgentState, stats chan<- StatsEvent) {
+func (a *Agent) Run(recordsChan chan<- *AvroDatum, stats chan<- StatsEvent) {
 	// fetch agent ip once. per DC/OS docs, changing a node IP is unsupported
-	var agentIp string = ""
+	a.AgentIP = ""
```

Comment:
unless you explicitly set `a.AgentIP` to empty string this is redundant. Strings are empty by default